packages/astro/src/core/config/schema.ts

```diff
@@ -330,12 +333,22 @@ export const AstroConfigSchema = z.object({
 						}),
 					])
 				),
+				domains: z
+					.record(
+						z.string(),
+						z
+							.string()
+							.url(
+								"The domain value must be a valid URL, and it has to start with 'https' or 'http'."
+							)
```

Comment:
I don't think this will filter out URLs that aren't using `https` or `http`, for the record. Probably fine, but we could add an additional validation if needed.